java/arcs/core/storage/StorageProxy.kt

```diff
@@ -412,9 +424,16 @@ class StorageProxy<Data : CrdtData, Op : CrdtOperationAtTime, T>(
         toResolve.forEach { it.complete(value) }
 
         when (priorState) {
-            ProxyState.AWAITING_SYNC -> notifyReady()
-            ProxyState.SYNC -> if (valueBefore != value) notifyUpdate(value)
-            ProxyState.DESYNC -> notifyResync()
+            ProxyState.AWAITING_SYNC -> {
+                notifyReady()
+                applyPostSyncModelOps()
+            }
+            ProxyState.SYNC -> notifyUpdate(value)
+            ProxyState.DESYNC -> {
+                notifyResync()
+                notifyUpdate(value)
```

Comment:
I don't think we want the call to `notifyUpdate`. Particles that want resync-specific behaviour will have that in the resync handler, and it's not a given that they would also want an update immediately after. Particles that don't care will already have `onResync` mapped to `onUpdate` (in `BaseParticle.kt`)

Reply:
The update sent here is only actually sent if the data is different from the last update that was sent, which maintains the implicit (explicit?) contract of `onUpdate`. I think it makes more sense for particle developers to not need to understand or care what a resync event is if they're really only curious about update events and reacting to them. Why would they care when they're synced and not synced?

Reply:
That's not how the API is intended to work, although I can see now there's a gap in the implementation.

You're right that most particles won't care; all they want to see is `onReady` followed by a stream of `onUpdate`s. For the particle-level API that's already what happens, but for handles there's nothing (prior to this CL) tying `onResync` to `onUpdate`.

Forcing a call to `notifyUpdate` here isn't the right long-term solution, however, because some particles may want specific resync behaviour that might not include the standard update flow, and we shouldn't force them to detect a resync-then-update just to avoid that update.

Not sure how best to fix this though; the Kotlin approach of attaching callbacks for handles (as opposed to overriding methods) means the BaseParticle redirect used for Particle.onResync won't work